pkg/azure/aks/machine_store.go

```diff
@@ -0,0 +1,273 @@
+	vmInfo := make(map[string]*VirtualMachineInfo)
+
+	opts := &armcompute.VirtualMachineScaleSetVMsClientListOptions{
+		Expand: to.StringPtr("instanceView"),
+	}
+	pager := m.azVMSSVmClient.NewListPager(rgName, vmssName, opts)
+	for pager.More() {
+		nextResult, err := pager.NextPage(m.context)
+		if err != nil {
+			return nil, ErrPageAdvanceFailure
+		}
+
+		for _, v := range nextResult.Value {
+			vmName, err := determineMachineName(v)
+			if err != nil {
+				m.logger.Error(err.Error())
+				continue
+			}
+
+			vmInfo[vmName] = &VirtualMachineInfo{
+				Name:            vmName,
+				Region:          to.String(v.Location),
+				OwningVMSS:      vmssName,
+				MachineTypeSku:  to.String(v.SKU.Name),
+				Priority:        priority,
+				OperatingSystem: osInfo,
+			}
+			m.logger.LogAttrs(m.context, slog.LevelDebug, "found machine information", slog.String("machineName", vmName))
+		}
+	}
+
+	return vmInfo, nil
+}
+
+func (m *MachineStore) getVmInfoFromResourceGroup(rgName, clusterName string) (map[string]*VirtualMachineInfo, error) {
+	vmInfoMap := make(map[string]*VirtualMachineInfo)
+
+	pager := m.azVMSSClient.NewListPager(rgName, nil)
+	for pager.More() {
+		nextResult, err := pager.NextPage(m.context)
+		if err != nil {
+			return nil, ErrPageAdvanceFailure
+		}
+
+		for _, v := range nextResult.Value {
+			vmssName := to.String(v.Name)
+			vmssPriority := determineMachineScaleSetPriority(v)
+			vmssOperationSystem := determineMachineScaleSetOperatingSystem(v)
+
+			vmInfo, err := m.getVmInfoFromVmss(rgName, vmssName, vmssPriority, vmssOperationSystem)
+			if err != nil {
+				return nil, err
+			}
+
+			for name, info := range vmInfo {
+				info.OwningCluster = clusterName
+				vmInfoMap[name] = info
+			}
+		}
+	}
+
+	return vmInfoMap, nil
+}
+
+func (m *MachineStore) setRegionListFromClusterList(rgList []*armcontainerservice.ManagedCluster) {
+	m.regionListLock.Lock()
+	defer m.regionListLock.Unlock()
+
+	locationSet := make(map[string]bool)
+	uniqueLocationList := []string{}
+
+	for _, v := range rgList {
+		locationSet[to.String(v.Location)] = true
+	}
+
+	for r := range locationSet {
+		uniqueLocationList = append(uniqueLocationList, r)
+	}
+
+	m.RegionList = uniqueLocationList
+}
+
+func (m *MachineStore) getClustersInSubscription() ([]*armcontainerservice.ManagedCluster, error) {
+	clusterList := []*armcontainerservice.ManagedCluster{}
+
+	pager := m.azAksClient.NewListPager(nil)
+	for pager.More() {
+		page, err := pager.NextPage(m.context)
+		if err != nil {
+			return nil, ErrPageAdvanceFailure
+		}
+		clusterList = append(clusterList, page.Value...)
+	}
+
+	return clusterList, nil
+}
+
+func (m *MachineStore) PopulateMachineStore() error {
+	startTime := time.Now()
+	m.logger.Info("populating machine store")
+
+	m.machineMapLock.Lock()
+	defer m.machineMapLock.Unlock()
+
+	clusterList, err := m.getClustersInSubscription()
+	if err != nil {
+		return err
+	}
+
+	go m.setRegionListFromClusterList(clusterList)
```

Comment:
I don't quite follow why you'd want to run this in a goroutine and update the region within the method. It seems like a bit of a smell and has potential for race conditions. 

Suggestion:

```go
m.RegionList := regionListFromClusterList(clusterList)
```

Or don't run the method in a goroutine.